rotkehlchen/api/rest.py

```diff
@@ -2110,64 +1808,49 @@ def _refresh_evm_accounts(self) -> dict[str, Any]:
 
         return OK_RESULT
 
-    def refresh_evm_accounts(self, async_query: bool) -> Response:
-        """
-        Check whether already added addresses have activity in other evm chains and if they do,
-        start tracking them in those chains too.
-        """
-        if async_query is True:
-            return self._query_async(command=self._refresh_evm_accounts)
-
-        response = self._refresh_evm_accounts()
-        result = response['result']
-        msg = response['message']
-        status_code = _get_status_code_from_async_response(response)
-
-        if result is None:
-            return api_response(wrap_in_fail_result(msg), status_code=status_code)
-
-        # success
-        result_dict = _wrap_in_result(result, msg)
-        return api_response(process_result(result_dict), status_code=status_code)
-
     def get_blockchain_accounts(self, blockchain: SupportedBlockchain) -> Response:
         with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
             data = self.rotkehlchen.get_blockchain_account_data(cursor, blockchain)
         return api_response(process_result(_wrap_in_result(data, '')), status_code=HTTPStatus.OK)
 
     @overload
-    def _add_single_blockchain_accounts(
+    @asynchronous()
+    def add_single_blockchain_accounts(
             self,
             chain: SUPPORTED_EVM_CHAINS,
             account_data: list[SingleBlockchainAccountData[ChecksumEvmAddress]],
     ) -> dict[str, Any]:
         ...
 
     @overload
-    def _add_single_blockchain_accounts(
+    @asynchronous()
+    def add_single_blockchain_accounts(
             self,
             chain: SUPPORTED_SUBSTRATE_CHAINS,
             account_data: list[SingleBlockchainAccountData[SubstrateAddress]],
     ) -> dict[str, Any]:
         ...
 
     @overload
-    def _add_single_blockchain_accounts(
+    @asynchronous()
```

Comment:
I changed it a bit, but now `async_query` argument has to be specified in the overloads and the return type of overloads became `Response`

Reply:
Yes, I don't mean that. What I mean if I'm correct is that you don't need the `@asynchronous` decorator in the methods that have the `@overload` since the definition is empty. You can remove it and leave it only in the method that has the actual behaviour